src/particles/diagnostics/ReducedBeamCharacteristics.cpp

```diff
@@ -36,6 +36,89 @@ namespace impactx::diagnostics
         // preparing access to particle data: AoS and SoA
         using PType = typename ImpactXParticleContainer::SuperParticleType;
 
+        // prepare reduction operations for minimum and maximum values from 6D phase space
+        amrex::ReduceOps<
+            amrex::ReduceOpMin, amrex::ReduceOpMax,  // x
+            amrex::ReduceOpMin, amrex::ReduceOpMax,  // y
+            amrex::ReduceOpMin, amrex::ReduceOpMax,  // t
+            amrex::ReduceOpMin, amrex::ReduceOpMax,  // px
+            amrex::ReduceOpMin, amrex::ReduceOpMax,  // py
+            amrex::ReduceOpMin, amrex::ReduceOpMax   // pt
+        > min_max_reduce_ops;
```

Comment:
Please combine with the reduce below, so we only need to iterate the particles once :)